data_structures/binary_tree/diff_views_of_binary_tree.py

```diff
@@ -0,0 +1,230 @@
+
+        >>> dfs([], 0, [])
+        None
+        """
+        if not root:
+            return
+
+        if depth == len(left_view):
+            left_view.append(root.val)
+
+        dfs(root.left, depth + 1, left_view)
+        dfs(root.right, depth + 1, left_view)
+
+    left_view: list = []
+    if not root:
+        return left_view
+    dfs(root, 0, left_view)
+    return left_view
+
+
+def binary_tree_top_side_view(root: TreeNode | None) -> list[int]:
+    """
+    Function returns the top side view of binary tree.
+
+    >>> binary_tree_top_side_view(None)
+    []
+    """
+    from collections import defaultdict
+
+    def bfs(root: TreeNode, top_view: list[int]) -> None:
+        """
+        A breadth first search traversal with defaultdict ds to append
+        the values of tree from top view
+
+        >>> bfs(TreeNode(5), [])
+        None
+        """
+        queue = [(root, 0)]
+        lookup = defaultdict(list)
+
+        while queue:
+            first = queue.pop(0)
+            node, hd = first
+            lookup[hd].append(node.val)
+
+            if node.left:
+                queue.append((node.left, hd - 1))
+            if node.right:
+                queue.append((node.right, hd + 1))
+
+        for key, val in sorted(lookup.items(), key=lambda each: each[0]):
+            top_view.append(val[0])
+
+    top_view: list = []
+    if not root:
+        return top_view
+    bfs(root, top_view)
+    return top_view
+
+
+def binary_tree_bottom_side_view(root: TreeNode | None) -> list[int]:
+    """
+    Function returns the bottom side view of binary tree
+
+    >>> binary_tree_bottom_side_view(None)
+    []
+    """
+    from collections import defaultdict
+
+    def bfs(root: TreeNode, bottom_view: list[int]) -> None:
+        """
+        A breadth first search traversal with defaultdict ds to append
+        the values of tree from bottom view
+
+        >>> bfs(TreeNode(5), [])
+        None
+        """
+        queue = [(root, 0)]
+        lookup = defaultdict(list)
+
+        while queue:
+            first = queue.pop(0)
+            node, hd = first
+            lookup[hd].append(node.val)
+
+            if node.left:
+                queue.append((node.left, hd - 1))
+            if node.right:
+                queue.append((node.right, hd + 1))
+
+        for key, val in sorted(lookup.items(), key=lambda each: each[0]):
+            bottom_view.append(val[-1])
+
+    bottom_view: list = []
+    if not root:
+        return bottom_view
+    bfs(root, bottom_view)
+    return bottom_view
+
+
+tree_1 = TreeNode(3)
+tree_1.left = TreeNode(9)
+tree_1.right = TreeNode(20)
+tree_1.right.left = TreeNode(15)
+tree_1.right.right = TreeNode(7)
+
+print(binary_tree_right_side_view(tree_1))  # Output: [3, 20, 7]
+print(binary_tree_left_side_view(tree_1))  # Output: [3, 9, 15]
+print(binary_tree_top_side_view(tree_1))  # Output: [9, 3, 20, 7]
+print(binary_tree_bottom_side_view(tree_1))  # Output: [9, 15, 20, 7]
```

Comment:
Could you please move these examples into a doctest? Thanks in advance...

Reply:
Thanks @CenTdemeern1 , I'm moving these examples in if block

Reply:
@kondekarshubham123 I asked if you could put these in a doctest, this is what doctests are for